testApp/cmd/testApp/main.go

```
@@ -0,0 +1,279 @@
+	gasPrice        = uint64(100000000000000)
+	gasLimit        = uint64(100000)
+	gasPerDataByte  = uint64(1500)
+)
+
+var network = [...]string{"Mainnet", "Testnet"}
+var status = [...]string{"Disabled", "Enabled"}
+
+const (
+	errOpenDevice         = "Couldn't open device"
+	errGetAppVersion      = "Couldn't get app version"
+	errGetConfig          = "Couldn't get configuration"
+	errGetAddress         = "Couldn't get address"
+	errGetBalanceAndNonce = "Couldn't get address balance and nonce"
+	errEmptyAddress       = "Empty address"
+	errInvalidAddress     = "Invalid receiver address"
+	errInvalidAmount      = "Invalid ERD amount"
+	errSigningTx          = "Signing error"
+	errSendingTx          = "Error sending tx"
+)
+
+type transaction struct {
+	Nonce     uint64 `json:"nonce"`
+	Value     string `json:"value"`
+	RcvAddr   string `json:"receiver"`
+	SndAddr   string `json:"sender"`
+	GasPrice  uint64 `json:"gasPrice,omitempty"`
+	GasLimit  uint64 `json:"gasLimit,omitempty"`
+	Data      string `json:"data,omitempty"`
+	Signature string `json:"signature,omitempty"`
+}
+
+type accountType struct {
+	Address string `json:"address"`
+	Nonce   uint64 `json:"nonce"`
+	Balance string `json:"balance"`
+}
+
+type accountMessage struct {
+	Account accountType `json:"account"`
+}
+
+// getSenderInfo returns the balance and nonce of an address
+func getSenderInfo(address string) (*big.Int, uint64, error) {
+	req, err := http.NewRequest(http.MethodGet,
+		fmt.Sprintf("%s/address/%s", observerHost, address), nil)
+	if err != nil {
+		return nil, 0, err
+	}
+	client := http.DefaultClient
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, 0, err
+	}
+	body, err := ioutil.ReadAll(resp.Body)
+	defer resp.Body.Close()
+	if err != nil {
+		return nil, 0, err
+	}
+	var balanceMsg accountMessage
+	err = json.Unmarshal(body, &balanceMsg)
+	if err != nil {
+		return nil, 0, err
+	}
+	balance, _ := big.NewInt(0).SetString(balanceMsg.Account.Balance, 10)
```

maybe should not ignore the error

---

Done.